### PR TITLE
Add various config options to the build documentation

### PR DESCRIPTION
**Pull request by fkleedorfer**
I find myself needing this from time to time and would appreciate being able to find it in the future using my favourite search engine.

EDIT: I was hoping to be able to skip the paperwork (create an issue etc.) for this small change.

### CODE REVIEW

**Comment by fkleedorfer** [comment]
@Aklakan great, I'll add that!

**Review by afs — CHANGES_REQUESTED** [review]

File: BUILD.md [hunk 1]
```diff
@@ -59,6 +59,21 @@ TDB, command line tools and Fuseki2.
...
+
+    mvn -DskipTests -Dmaven.javadoc.skip=true clean install
+
+Build only a specific module (e.g. `jena-arq`) and its dependencies
+
+    mvn -pl :jena-arq -am install
+
+Also useful:
+
+`-Denforcer.skip` If the maven version is too old (e.g. server deployment), skipping the enforcer may still result in a successful build
```
[the remark this answers is not in the document]
Comment (fkleedorfer) [reply BUILD.md:72]: I cannot really defend this addition. @Aklakan can you?

File: BUILD.md [hunk 2]
```diff
@@ -59,6 +59,21 @@ TDB, command line tools and Fuseki2.
 Once the whole of Jena has been built once, individual modules can be incrementally
 built using maven in their module directory.
 
+To build the whole project, quickly and dirtily, i.e, skipping tests (but building them because that's required) and skipping javadoc generation (because it might fail):
```
[the remark this answers is not in the document]
Comment (fkleedorfer) [reply BUILD.md]: as a side note: I started skipping javadoc generation because it failed

[the remark this answers is not in the document]
Comment (fkleedorfer) [reply BUILD.md]:
```
$ mvn -version
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: C:\DATA\DEV\programs\apache-maven-3.8.1
Java version: 15.0.1, vendor: AdoptOpenJDK, runtime: C:\DATA\DEV\programs\AdoptOpenJDK15
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

```
$ mvn -DskipTests clean install
[...]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Jena 4.6.0-SNAPSHOT:
[INFO]
[INFO] Apache Jena ........................................ SUCCESS [ 24.831 s]
[INFO] Apache Jena - Shadowed external libraries .......... SUCCESS [  3.731 s]
[INFO] Apache Jena - IRI .................................. SUCCESS [  8.255 s]
[INFO] Apache Jena - Base Common Environment .............. SUCCESS [ 13.956 s]
[INFO] Apache Jena - Core ................................. SUCCESS [ 39.787 s]
[INFO] Apache Jena - ARQ .................................. SUCCESS [ 55.033 s]
[INFO] Apache Jena - SHACL ................................ SUCCESS [  8.670 s]
[INFO] Apache Jena - ShEx ................................. SUCCESS [  7.745 s]
[INFO] Apache Jena - RDF Connection ....................... FAILURE [  4.056 s]
[INFO] Apache Jena - TDB1 (Native Triple Store) ........... SKIPPED
[INFO] Apache Jena - DBOE Database Operation Environment .. SKIPPED
[INFO] Apache Jena - DBOE Base ............................ SKIPPED
[INFO] Apache Jena - DBOE Transactions .................... SKIPPED
[INFO] Apache Jena - DBOE Indexes ......................... SKIPPED
[INFO] Apache Jena - DBOE Index test suite ................ SKIPPED
[INFO] Apache Jena - DBOE Transactional Datastructures .... SKIPPED
[INFO] Apache Jena - DBOE Storage ......................... SKIPPED
[INFO] Apache Jena - DBOE TDB2 (Native Triple Store) ...... SKIPPED
[INFO] Apache Jena - Libraries POM ........................ SKIPPED
[INFO] Apache Jena - Command line tools ................... SKIPPED
[INFO] Apache Jena - SPARQL Text Search ................... SKIPPED
[INFO] Apache Jena - Fuseki - A SPARQL 1.1 Server ......... SKIPPED
[INFO] Apache Jena - Fuseki Core Engine ................... SKIPPED
[INFO] Apache Jena - Fuseki Data Access Control ........... SKIPPED
[INFO] Apache Jena - Fuseki Server Main ................... SKIPPED
[INFO] Apache Jena - Fuseki Server Jar .................... SKIPPED
[INFO] Apache Jena - Fuseki Webapp ........................ SKIPPED
[INFO] Apache Jena - Fuseki UI ............................ SKIPPED
[INFO] Apache Jena - Fuseki WAR File ...................... SKIPPED
[INFO] Apache Jena - Fuseki Server Standalone Jar ......... SKIPPED
[INFO] Apache Jena - Fuseki Docker Tools .................. SKIPPED
[INFO] Apache Jena - Fuseki Binary Distribution ........... SKIPPED
[INFO] Apache Jena - GeoSPARQL Engine ..................... SKIPPED
[INFO] Apache Jena - Fuseki with GeoSPARQL Engine ......... SKIPPED
[INFO] Apache Jena - Integration Testing .................. SKIPPED
[INFO] Apache Jena - Distribution ......................... SKIPPED
[INFO] Apache Jena - Security Permissions ................. SKIPPED
[INFO] Apache Jena - Extras ............................... SKIPPED
[INFO] Apache Jena - Extras - Query Builder ............... SKIPPED
[INFO] Apache Jena - CommonsRDF for Jena .................. SKIPPED
[INFO] Apache Jena - Code Examples ........................ SKIPPED
[INFO] Apache Jena - JDBC Parent .......................... SKIPPED
[INFO] Apache Jena - JDBC Core API ........................ SKIPPED
[INFO] Apache Jena - JDBC Remote Endpoint Driver .......... SKIPPED
[INFO] Apache Jena - JDBC In-Memory Driver ................ SKIPPED
[INFO] Apache Jena - JDBC TDB Driver ...................... SKIPPED
[INFO] Apache Jena - JDBC Driver Bundle ................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:47 min
[INFO] Finished at: 2022-07-07T16:17:45+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.4.0:jar (attach-javadocs) on project jena-rdfconnection: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1 - C:\DATA\DEV\workspace\jena\jena-rdfconnection\src\main\java\org\apache\jena\rdflink\RDFLink.java:22: error: package java.net.http is not visible
[ERROR] import java.net.http.HttpClient;
[ERROR]                ^
[ERROR]   (package java.net.http is declared in module java.net.http, but module org.apache.jena.rdfconnection does not read it)
[ERROR] C:\DATA\DEV\workspace\jena\jena-rdfconnection\src\main\java\org\apache\jena\rdfconnection\RDFConnection.java:24: error: package java.net.http is not visible
[ERROR] import java.net.http.HttpClient;
[ERROR]                ^
[ERROR]   (package java.net.http is declared in module org.apache.jena.arq, which does not export it)
[ERROR] C:\DATA\DEV\workspace\jena\jena-rdfconnection\src\main\java\org\apache\jena\rdfconnection\LibSec.java:24: error: package java.net.http is not visible
[ERROR] import java.net.http.HttpClient;
[ERROR]                ^
[ERROR]   (package java.net.http is declared in module org.apache.jena.arq, which does not export it)
[ERROR] C:\DATA\DEV\workspace\jena\jena-rdfconnection\src\main\java\org\apache\jena\rdfconnection\RDFConnectionRemoteBuilder.java:21: error: package java.net.http is not visible
[ERROR] import java.net.http.HttpClient;
[ERROR]                ^
[ERROR]   (package java.net.http is declared in module org.apache.jena.arq, which does not export it)
[ERROR] C:\DATA\DEV\workspace\jena\jena-rdfconnection\src\main\java\org\apache\jena\rdflink\RDFLinkHTTPBuilder.java:23: error: package java.net.http is not visible
[ERROR] import java.net.http.HttpClient;
[ERROR]                ^
[ERROR]   (package java.net.http is declared in module org.apache.jena.arq, which does not export it)
[ERROR] C:\DATA\DEV\workspace\jena\jena-rdfconnection\src\main\java\org\apache\jena\rdflink\RDFLinkHTTP.java:21: error: package java.net.http is not visible
[ERROR] import java.net.http.HttpClient;
[ERROR]                ^
[ERROR]   (package java.net.http is declared in module org.apache.jena.arq, which does not export it)
[ERROR] C:\DATA\DEV\workspace\jena\jena-rdfconnection\src\main\java\org\apache\jena\rdflink\RDFLinkFuseki.java:21: error: package java.net.http is not visible
[ERROR] import java.net.http.HttpClient;
[ERROR]                ^
[ERROR]   (package java.net.http is declared in module org.apache.jena.arq, which does not export it)
[ERROR]
[ERROR] Command line was: cmd.exe /X /C "C:\DATA\DEV\programs\AdoptOpenJDK15\bin\javadoc.exe @options @packages @argfile"
[ERROR]
[ERROR] Refer to the generated Javadoc files in 'C:\DATA\DEV\workspace\jena\jena-rdfconnection\target\apidocs' dir.
[ERROR]
[...]
```

I can file an issue if you like - I was just pressed for time when I first ran into it...

**Comment by afs** [comment]
> Java version: 15.0.1

I'm not seeing this in OpenJDK Java11, Java17 or Java19 (phew!).

File: BUILD.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aklakan) [reply BUILD.md:72]: The latest maven that comes with ubuntu 22.04 is presently [3.6.3-5](https://packages.ubuntu.com/search?keywords=maven) - 
When building Jena with that version I get
```
[WARNING] Rule 3: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
Detected Maven Version: 3.6.3 is not in the allowed range 3.8.1.
```
However with -Denforcer.skip it builds just fine for me. Of course mileage may vary.

[the remark this answers is not in the document]
Comment (afs) [reply BUILD.md:72]: I don't think we should encourage this - there will be problems when submitting a PR.

[the remark this answers is not in the document]
Comment (fkleedorfer) [reply BUILD.md:72]: ok, I'll remove the hint

**Review by afs — APPROVED** [review]

**Pull request merged**